### PR TITLE
Day 2, 3 problems

### PR DESCRIPTION
**Pull request by raghavsrm**

### CODE REVIEW

File: src/com/raghav/javabasics/CompareTwoNumbersReturnMax.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.raghav.javabasics;
+//Given two numbers, write a program to compare the numbers and return the max among them.
+
+public class CompareTwoNumbersReturnMax {
+    public static void main(String args[]){
+        CompareTwoNumbersReturnMax max =new CompareTwoNumbersReturnMax();
+        System.out.println(max.findMax(5, 10));
+        System.out.println(max.findMax(0,2));
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/CompareTwoNumbersReturnMax.java:8]: What about the case where num1=num2? Add test for that case as well.
And the case where you pass negative numbers.

File: src/com/raghav/javabasics/Counter.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package com.raghav.javabasics;
+// words, char, lines
+
+public class Counter {
+    public static void main(String[] args) {
+      String paragraph = "Change an Array\n" +
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/Counter.java:6]: What happens if your input string is empty or a null value?
String paragraph=“”; or
String paragraph=null;

File: src/com/raghav/javabasics/Counter.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+              "new\n" +
+              "out\n" +
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/Counter.java:8]: What happens when paragraph=“SingleWord”? Please add tests cases for the same

File: src/com/raghav/javabasics/Counter.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+        for (int i=0;i<temp.length;i++)
+        System.out.println(temp[i]);
+    }
+    public int words(String paragraph){
+        int wordCount=1;
+        for(int i=0;i<paragraph.length();i++){
+
+            char c = paragraph.charAt(i);
+            if(c == ' ' || c == '\n'){
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/Counter.java:22]: What happens if there are some special characters in the string? Add testcase for the same.
String paragraph=“this,is.also.a.sentence”;
What are all the delimiters you are considering?

File: src/com/raghav/javabasics/Counter.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            charCount = charCount+1;
+        }
+        return charCount;
+    }
+    public int lines(String paragraph){
+        int lineCount=1;
+        for(int i=0;i<paragraph.length();i++){
+            char c = paragraph.charAt(i);
+            if(c == '\n'){
+                lineCount = lineCount+1;
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/Counter.java:42]: What happens if there is a ‘\n’ at the end of the string as last character without any words, do we still consider it as a new line?

File: src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+//Given an array of numbers, write a program to identify all numbers divisible by 2,
+// divisible by 3, divisible by 5, divisible by 7, divisible by 5 and divisible by 7.
+
+public class DisplayArrayNumbersAreDivisible {
+    public static void main(String[] args) {
+
+        DisplayArrayNumbersAreDivisible result = new DisplayArrayNumbersAreDivisible();
+        result.divisibilityChecker(new int[]{1,2,3,4,5,6,7,8,9,35});
+        result.divisibilityChecker(new int[]{});
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java:12]: What about 0 and negative numbers?

File: src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    public void divisibilityChecker(int[] myArray) {
+        int[] divNumArr = {2, 3, 5, 7, 35};
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java:16]: Why 35? Where did this number come from?

File: src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
+package com.raghav.javabasics;
+
+//Given an array of numbers, write a program to identify all the even indexed numbers and odd indexed numbers.
+
+public class FindAllEvenOddIndexedNumbersInArray {
+    public static void main(String[] args) {
+        FindAllEvenOddIndexedNumbersInArray result = new FindAllEvenOddIndexedNumbersInArray();
+        result.countIndices(new int[]{5,-5,6,2,1,99,23});
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java:8]: What happens when array is empty or null?

File: src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+        int oddCount =0;
+        for( int i=0; i< myArr.length; i++){
+            if(i == 0 || i % 2 ==0){
+                evenCount = evenCount + 1;
+            }
+            else{
+                oddCount = oddCount + 1;
+            }
+        }
+        System.out.println("Count of Even Indexes : "+evenCount+" Count of Odd Indexes : "+oddCount);
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java:22]: Program is not to count number of odd indexes and even indexed, it is to return numbers at odd and even indexed.

File: src/com/raghav/javabasics/FindElementInArray.java [hunk 10]
```diff
@@ -1,35 +1,20 @@
...
-        int[] myArray = new int[size];
-        System.out.println("Enter the elements of the array ::");
-
-        for (int i = 0; i < size; i++) {
-            myArray[i] = sc.nextInt();
-        }
-//        FindElementInArray m = new FindElementInArray();
-        System.out.println("Enter integer: ");
+        FindElementInArray result = new FindElementInArray();
+        result.findElement(new int[]{5,-5,6,2,1,99,23}, 99);
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/FindElementInArray.java:7]: What happens if array is empty?

File: src/com/raghav/javabasics/FindEvenOddInArray.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+        int oddCount =0;
+        for( int i=0; i< myArray.length; i++){
+            if(myArray[i]%2 == 0){
+                evenCount +=1;
+            }
+            else{
+                oddCount+=1;
+            }
+        }
+        System.out.println("Even Number Count: "+evenCount+" Odd Number Count: "+oddCount);
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/FindEvenOddInArray.java:22]: Program is not to find how many odd and even numbers but to print/return them

File: src/com/raghav/javabasics/FindEvenOddInArray.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
+package com.raghav.javabasics;
+
+// Given an array of numbers, write a program to find the number of odd and number of even numbers.
+
+public class FindEvenOddInArray {
+    public static void main(String[] args) {
+        FindEvenOddInArray result = new FindEvenOddInArray();
+        result.findEvenOddNum(new int[]{5,-5,6,2,1,99,23});
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/FindEvenOddInArray.java:8]: What if array is empty?

File: src/com/raghav/javabasics/FindMaxNumberInArray.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
+package com.raghav.javabasics;
+
+//Given an array of numbers, write a program to identify the max number in the list.
+
+public class FindMaxNumberInArray {
+    public static void main(String[] args) {
+        FindMaxNumberInArray result = new FindMaxNumberInArray();
+        result.findMax(new int[]{1,5,858,2,5,4,5,2158,1,587,4});
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/FindMaxNumberInArray.java:8]: What if the max number is duplicate in the array? Do you return it once or twice?

File: src/com/raghav/javabasics/FindMaxNumberInArray.java [hunk 14]
```diff
@@ -0,0 +1,29 @@
+package com.raghav.javabasics;
+
+//Given an array of numbers, write a program to identify the max number in the list.
+
+public class FindMaxNumberInArray {
+    public static void main(String[] args) {
+        FindMaxNumberInArray result = new FindMaxNumberInArray();
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/FindMaxNumberInArray.java:7]: What if array is empty?

File: src/com/raghav/javabasics/FindMinNumberInArray.java [hunk 15]
```diff
@@ -0,0 +1,34 @@
+package com.raghav.javabasics;
+
+//Given an array of numbers, write a program to identify the min number in the list.
+
+public class FindMinNumberInArray {
+    public static void main(String[] args) {
+        FindMinNumberInArray result = new FindMinNumberInArray();
+        result.findMin(new int[]{53,11,544,15,55,21,54,5});
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/FindMinNumberInArray.java:8]: What about negative number?

File: src/com/raghav/javabasics/FindMinNumberInArray.java [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+        result.findMin(new int[]{});
+    }
+
+    public void findMin(int[] myArray) {
+        if (myArray.length == 0) {
+            System.out.println("Null");
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/FindMinNumberInArray.java:14]: Have a more descriptive error message.

File: src/com/raghav/javabasics/RomanToNumber.java [hunk 17]
```diff
@@ -1,9 +1,14 @@
 package com.raghav.javabasics;
 
-import java.util.*;
+public class RomanToNumber{
+    public static void main(String args[])
+    {
+        RomanToNumber rn = new RomanToNumber();
+        System.out.println("Integer form of Roman Numeral is: "+ rn.romanToDecimal("VXIII"));
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/RomanToNumber.java:7]: What if we send empty string or null?

File: src/com/raghav/javabasics/SumOfAllEvenAndOddNumbersInAnArray.java [hunk 18]
```diff
@@ -0,0 +1,24 @@
+package com.raghav.javabasics;
+
+//Given an array of numbers, write a program to find the sum of all even numbers and sum of all odd numbers.
+
+public class SumOfAllEvenAndOddNumbersInAnArray {
+    public static void main(String[] args) {
+        SumOfAllEvenAndOddNumbersInAnArray result = new SumOfAllEvenAndOddNumbersInAnArray();
+        result.sumOfEvenOdd(new int[]{2,1,5,4,3,6,8,4,5});
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/SumOfAllEvenAndOddNumbersInAnArray.java:8]: What if i have negative numbers?

File: src/com/raghav/javabasics/SumOfAllNumbersInArray.java [hunk 19]
```diff
@@ -0,0 +1,20 @@
+package com.raghav.javabasics;
+
+//Given an array of numbers, write a program to return the sum of all the numbers.
+
+public class SumOfAllNumbersInArray {
+    public static void main(String[] args) {
+        SumOfAllNumbersInArray result = new SumOfAllNumbersInArray();
+        result.sumOfAllNums(new int[]{1,2,3,5,4,6,2});
+        result.sumOfAllNums(new int[]{});
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/SumOfAllNumbersInArray.java:9]: Will anything be printed when array is empty? Please add appropriate print statements indicating array is empty. This applies to all the other programs as well

File: src/com/raghav/javabasics/SumOfAllNumbersInArray.java [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+    }
+
+    public void sumOfAllNums(int[] myArray){
+        int swapValue1; int swapValue2= 0;
+        for (int i = 0; i < myArray.length; i++) {
+            swapValue1 = myArray[i];
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/SumOfAllNumbersInArray.java:15]: Why do we have swapValue1 variable? Wouldnt line 16 be enough?

File: src/com/raghav/javabasics/CheckNumberIsBinary.java [hunk 21]
```diff
@@ -0,0 +1,28 @@
+package com.raghav.javabasics;
+// Write a java program to check whether given number is binary or not. A binary number is a number which contains only 0 or 1.
+
+public class CheckNumberIsBinary {
+
+    public static void main(String args[]) {
```
Comment (soumyamandadi) [inline src/com/raghav/javabasics/CheckNumberIsBinary.java:6]: For all the programs, can you add the run output in the description to identify what the output of each programs are?

Comment (raghavsrm) [reply src/com/raghav/javabasics/CheckNumberIsBinary.java:6]: output: binary
not binary
not binary
binary
binary

File: src/com/raghav/javabasics/Counter.java (hunk 5 above)
Comment (raghavsrm) [reply src/com/raghav/javabasics/Counter.java:43]: working on it.

File: src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java (hunk 6 above)
Comment (raghavsrm) [reply src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java:12]: output:
Array Numbers Divisible by 2: [2, 4, 6, 8, 0, 0, 0, 0, 0, 0]
Array Numbers Divisible by 3: [3, 6, 9, 0, 0, 0, 0, 0, 0, 0]
Array Numbers Divisible by 5: [5, 35, 0, 0, 0, 0, 0, 0, 0, 0]
Array Numbers Divisible by 7: [7, 35, 0, 0, 0, 0, 0, 0, 0, 0]
Array Numbers Divisible by 35: [35, 0, 0, 0, 0, 0, 0, 0, 0, 0]
Array Numbers Divisible by 2: []
Array Numbers Divisible by 3: []
Array Numbers Divisible by 5: []
Array Numbers Divisible by 7: []
Array Numbers Divisible by 35: []

File: src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java (hunk 7 above)
Comment (raghavsrm) [reply src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java:16]: number divisible by 5 and 7

File: src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java (hunk 6 above)
Comment (mspranav) [reply src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java:12]: Negative Numbers?

File: src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java (hunk 7 above)
Comment (mspranav) [reply src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java:16]: Try to incorporate 5 and 7 in the code for 5 or 7.

if divisible by 5:
    if divisible by 7:
        both
    else:
        only 5
elif by 7:
     only 7

File: src/com/raghav/javabasics/CompareTwoNumbersReturnMax.java [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+        max.findMax(0, 40);
+    }
+
+    public void findMax(int num1, int num2){
+        if(num1==num2)
+            System.out.println("Both are equal");
+            else if (num1 > num2)
+                System.out.println(num1 + " is greater");
+            else
+                System.out.println(num2+" is greater");
```
Comment (mspranav) [inline src/com/raghav/javabasics/CompareTwoNumbersReturnMax.java:20]: Nit:  Please do follow brackets and indentation.

File: src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+        int count1=0;
+        int count2=0;
+        int[] evenCountArr = new int[myArr.length/2+1 ];
+        int[] oddCountArr = new int[myArr.length/2 ];
+
+        for( int i=0; i< myArr.length; i++){
+            if(i == 0 || i % 2 ==0){
+                evenCount = evenCount + 1;
+                evenCountArr[count1]=myArr[i];
+                count1+=1;
```
Comment (mspranav) [inline src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java:31]: Nit: either use ```+=``` or `count = count + 1`

File: src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+        }
+        int evenCount=0;
+        int oddCount =0;
+        int count1=0;
+        int count2=0;
+        int[] evenCountArr = new int[myArr.length/2+1 ];
+        int[] oddCountArr = new int[myArr.length/2 ];
+
+        for( int i=0; i< myArr.length; i++){
+            if(i == 0 || i % 2 ==0){
```
Comment (mspranav) [inline src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java:28]: `0 % 2 = 0`

File: src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+                oddCountArr[count2]=myArr[i];
+                count2+=1;
+            }
+        }
+        for(int i=0; i<count1;i++){
+            if(i==0){
+                System.out.print("Even indices list: ");
+            }
+            System.out.print(evenCountArr[i]+" ");
+        }
```
Comment (mspranav) [inline src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java:44]: what if the print stmt is outside the loop and then print each index from the loop?

File: src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java [hunk 26]
```diff
@@ -0,0 +1,52 @@
...
+        for(int i=0; i<count2;i++){
+            if(i==0){
+                System.out.print("   Odd indices list: ");
+            }
+            System.out.print(oddCountArr[i]+" ");
+        }    }
+}
+
```
Comment (mspranav) [inline src/com/raghav/javabasics/FindAllEvenOddIndexedNumbersInArray.java:52]: what if the print stmt is outside the loop and then print each index from the loop?

File: src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java (hunk 6 above)
Comment (raghavsrm) [reply src/com/raghav/javabasics/DisplayArrayNumbersAreDivisible.java:12]: Added in the latest commit

File: src/com/raghav/javabasics/FindMaxNumberInArray.java (hunk 14 above)
Comment (raghavsrm) [reply src/com/raghav/javabasics/FindMaxNumberInArray.java:7]: Added in the latest commit

**Review by raghavsrm — COMMENTED** [review]
Added in the latest commit

**Review by shashanksurya — APPROVED** [review]
LGTM!

**Pull request merged**